lesson13.md

```diff
@@ -0,0 +1,303 @@
+myInputElement.placeholder = "Amount";
+```
+
+---
+
+### Setting onclick in javascript?
+
+HTML
+
+```html
+<button id="myButton" onclick="myFunction()">Click me!</button>
+```
+
+How can we set the `onclick` property from JavaScript?
+
+```js
+let myButtonElement = document.getElementById("myButton");
+// ???
+```
+
+---
+
+### Variables pointing to functions
+
+So far, we learned that variables can point to values:
+
+```js
+let age = 42;
+let name = "Otto";
+let isHappy = true;
+let address = {
+    street: "Ottostr.",
+    city: "Berlin"
+};
+```
+
+---
+
+### Functions can be values, too
+
+Variables can also point to functions:
+
+```js
+let myFunction = function() {
+    console.log("this function was called");
+};
+// myFunction is a variable pointing to our function
+myFunction(); // calls the function above
+```
+
+The function above is also called an "anonymous" function, because the function
+itself has no name.
+
+---
+
+### Syntax comparison
+
+```js
+function myFunction(name) {
+    console.log("Hi " + name);
+}
+let myFunction2 = function(name) {
+    console.log("Hi " + name);
+}
+myFunction("Owen"); // "Hi Owen"
+myFunction2("Harald"); // "Hi Harald"
+```
+
+---
+
+HTML
+
+```html
+<button id="myButton">Click me!</button>
+```
+
+How can we set the `onclick` property from JavaScript?
+
+```js
+let myButtonElement = document.getElementById("myButton");
+myButtonElement.onclick = function() {
+    console.log("My button was clicked");
+}
+```
+---
+
+## Task
+
+Create a new page with two buttons with the text "red" and "blue", and a div element with the text "Change my background".
+
+Now in Javascript, add `onclick` handlers on these buttons, and when click on them the background of the div changes to that color.
+
+---
+
+### Solution
+
+```html
+<button id='red-button'>red</button>
+<button id='blue-button'>blue</button>
+<div id='text'>Change my background</div>
+```
+
+---
+
+### Solution
+
+
+```js
+const redButton = document.getElementById('red-button');
+const blueButton = document.getElementById('blue-button');
+const divElement = document.getElementById('text');
+
+
+redButton.onclick = function(){
+  divElement.style.backgroundColor = 'red';
+}
+
+blueButton.onclick = function(){
+  divElement.style.backgroundColor = 'blue';
+}
+
+
+```
+
+---
+
+So far, we can get elements from HTML and manipulate them.
+
+* We can set and get properties
+* properties can be numbers, strings, booleans, object, functions...
+* But what if we want to create a new element that's not in HTML yet?
+
+---
+
+<!-- .slide: id="create-element" -->
+
+### document.createElement
+
+Basic usage:
+```js
+let myDiv = document.createElement("div");
+```
+
+However, we need to do a bit more configuration to have something useful.
+
+---
+
+```js
+let myDiv = document.createElement("div"); // 1
+myDiv.textContent = "hello";               // 2
+document.body.appendChild(myDiv);          // 3
+// <div>hello</div> has been added to the page!
+```
+
+---
```

Comment:
Can we merge these two slides? Isn't it annoying to switch between two slides to see the code + explanation?